mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreRightManagerTest.java

```diff
@@ -51,238 +51,372 @@
+                .isTrue();
+        }
+
+        @Test
+        void isReadWriteShouldReturnTrueWhenSystemUserHasDeleteMessagesRightOnMailboxAndFlagsContainDeletedFlag() throws Exception {
+            Flags flags = new Flags(Flags.Flag.DELETED);
+            Mailbox mailbox = new Mailbox(MailboxPath.forUser(BOB, MailboxConstants.INBOX), UID_VALIDITY);
+            mailbox.setACL(new MailboxACL(new MailboxACL.Entry(MailboxFixture.ALICE.asString(), Right.DeleteMessages)));
+
+            assertThat(storeRightManager.isReadWrite(aliceSession, mailbox, flags))
+                .isTrue();
+        }
+
+        @Test
+        void isReadWriteShouldReturnTrueWhenSystemUserHasDeleteMessagesRightOnMailboxButFlagsDoesNotContainDeletedFlag() throws Exception {
+            Flags flags = new Flags();
+            Mailbox mailbox = new Mailbox(MailboxPath.forUser(BOB, MailboxConstants.INBOX), UID_VALIDITY);
+            mailbox.setACL(new MailboxACL(new MailboxACL.Entry(MailboxFixture.ALICE.asString(), Right.DeleteMessages)));
+
+            assertThat(storeRightManager.isReadWrite(aliceSession, mailbox, flags))
+                .isTrue();
+        }
+
+        @Test
+        void isReadWriteShouldReturnTrueWhenSystemUserHasWriteSeenFlagRightOnMailboxAndFlagsContainSeenFlag() throws Exception {
+            Flags flags = new Flags(Flags.Flag.SEEN);
+            Mailbox mailbox = new Mailbox(MailboxPath.forUser(BOB, MailboxConstants.INBOX), UID_VALIDITY);
+            mailbox.setACL(new MailboxACL(new MailboxACL.Entry(MailboxFixture.ALICE.asString(), Right.WriteSeenFlag)));
+
+            assertThat(storeRightManager.isReadWrite(aliceSession, mailbox, flags))
+                .isTrue();
+        }
+
+        @Test
+        void isReadWriteShouldReturnTrueWhenSystemUserHasWriteSeenFlagRightOnMailboxAndFlagsDoesNotContainSeenFlag() throws Exception {
+            Flags flags = new Flags();
+            Mailbox mailbox = new Mailbox(MailboxPath.forUser(BOB, MailboxConstants.INBOX), UID_VALIDITY);
+            mailbox.setACL(new MailboxACL(new MailboxACL.Entry(MailboxFixture.ALICE.asString(), Right.WriteSeenFlag)));
+
+            assertThat(storeRightManager.isReadWrite(aliceSession, mailbox, flags))
+                .isTrue();
+        }
+
+        @Test
+        void isReadWriteShouldReturnTrueWhenSystemUserHasWriteRightOnMailboxAndFlagsContainAnsweredFlag() throws Exception {
+            Flags flags = new Flags(Flags.Flag.ANSWERED);
+            Mailbox mailbox = new Mailbox(MailboxPath.forUser(BOB, MailboxConstants.INBOX), UID_VALIDITY);
+            mailbox.setACL(new MailboxACL(new MailboxACL.Entry(MailboxFixture.ALICE.asString(), Right.Write)));
+
+            assertThat(storeRightManager.isReadWrite(aliceSession, mailbox, flags))
+                .isTrue();
+        }
+
+        @Test
+        void isReadWriteShouldReturnTrueWhenSystemUserDoesNotHaveInsertOrPerformExpungeRightOnMailboxAndNullFlag() throws Exception {
```

Comment:
There is no such things as a **SystemUser**  plean prefer the **WhenSystemSession** wording...